webapp/packages/plugin-data-viewer-result-set-grouping/src/useGroupingDataModel.ts

```diff
@@ -90,6 +109,14 @@ export function useGroupingDataModel(
         };
       },
       async ({ columns, functions, sourceResultId }) => {
+        const prevState = prevStateRef.current;
+
+        if (columns == prevState.columns && functions == prevState.functions && sourceResultId == prevState.sourceResultId) {
```

Comment:
I'm not sure about this. Lets sync with the team and figure out wether we want to load new data each time we open grouping panel